lib/ansible/module_utils/vmware.py

```diff
@@ -29,20 +30,41 @@ class TaskError(Exception):
     pass
 
 
-def wait_for_task(task):
+def wait_for_task(task, max_backoff=64, timeout=3600):
+    """Function to wait for given task using exponential back-off algorithm.
+
+    Args:
+        task: VMware task object
+        max_backoff: Maximum amount of sleep time in seconds
+        timeout: Timeout for the given task in seconds
+
+    Returns: TaskError for failure or True with result for successful task.
```

Comment:
You should state `Raises: TaskError` separately, because it's not a return value